packages/plugin/src/theia.d.ts

```diff
@@ -3140,7 +3140,7 @@ export module '@theia/plugin' {
         /**
          * The icon path or {@link ThemeIcon} for the terminal.
          */
-        iconPath?: ThemeIcon;
+        iconPath?: string | ThemeIcon | Uri | { light: Uri | string, dark: Uri | string };
```

Comment:
Suggestion: In vscode, both of these `iconPath` definitions are defined as:

```ts
iconPath?: Uri | { light: Uri; dark: Uri } | ThemeIcon;
```

Currently, both of them are marked as `partial` by the comparator report.

Reply:
Does this mean that the implementation in this PR is a partial implementation over what is done in VS Code?

I was having trouble implementing the vscode definition for `iconPath` and change the icon addressing from URI to hosted address. Originally, `iconPath` uses `ThemeIcon` only for this reason. `ThemeIcon` are essentially codicons and thus the source of the icon is in the application itself. Using Uri means that the source for the icon can be references "externally". In this case the browser version does not like it referencing a specific file in the computer for security reasons. So the reference address has to be of the form "hostedPlugins/plugin/address". This is different on how URI handles addresses. So at the end using a URI to pass the relative address ends causing errors.

I constrained myself to finish this PR in a couple days before the end of my internship. The current solution is not ideal, and I would prefer a better one to be used, but implementing one would have taken more time.

There are some paths that can be taken to do this:
1. Changing how URI works and are defined. This would also streamline a bit of the process in Themables for the git plugin. (Another PR I worked on).
2. Change how the icon addressing information is passed so that you can pass a "hostedPlugin" address instead of just URI or ThemeIcons.